scripts/install_dependencies/install_openvino_dependencies.sh

```diff
@@ -16,260 +16,344 @@
+        *) break ;;
+    esac
+    shift
+done
+
+# No components selected - install all
+if [ ${#comp[@]} -eq 0 ]; then
+    comp=(${all_comp[@]})
 fi
 
-params=$@
+#echo "DBG:
+#    interactive=${interactive}
+#    extra=${extra}
+#    selftest=${selftest}
+#    print=${print}
+#    comp=${comp[*]}"
 
-yes_or_no_ffmpeg() {
-    if [ "$params" == "-y" ]; then
-        return 0
-    fi
+#===================================================================================================
+#  Selftest
 
-    while true; do
-        read -p "Add third-party RPM Fusion repository and install FFmpeg package (y/n): " yn
-        case $yn in
-            [Yy]*) return 0  ;;
-            [Nn]*) return  1 ;;
-        esac
+if [ -n "$selftest" ] ; then
+    for image in centos:7 ; do
+#    for image in centos:7 ubuntu:18.04 ubuntu:20.04 ; do
+#        for opt in  "-h" "-p" "-e -p" "-n" "-n -e" "-y" "-y -e" ; do
+#        for opt in  "-y" "-y -e" ; do
+        for opt in  "-y -e" ; do
+            echo "||"
+            echo "|| Test $image / '$opt'"
+            echo "||"
+            SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+            docker run -it --rm \
+                --volume ${SCRIPT_DIR}:/scripts:ro,Z  \
+                --volume yum-cache:/var/cache/yum \
+                --volume apt-cache:/var/cache/apt/archives \
+                -e DEBIAN_FRONTEND=noninteractive \
+                $image \
+                bash /scripts/${0##*/} $opt --keepcache
+            echo "||"
+            echo "|| Completed: $image / '$opt'"
+            echo "||"
+        done
     done
-}
+    echo "Self test finished, to remove temporary docker volumes run:
+        'docker volume rm yum-cache apt-cache'"
+    exit 0
+fi
 
-yes_or_no_gst_bad_ugly() {
-    if [ "$params" == "-y" ]; then
-        return 0
-    fi
+#===================================================================================================
+# OS detection
 
-    while true; do
-        read -p "Add third-party RPM Epel, Nux, Fusion, Forensics repositories and install dependencies for GStreamer Bad & Ugly Plugins (y/n): " yn
-        case $yn in
-            [Yy]*) return 0  ;;
-            [Nn]*) return  1 ;;
-        esac
-    done
-}
-
-if [ -f /etc/lsb-release ]; then
-    # Ubuntu
-    echo
-    echo "This script installs the following OpenVINO 3rd-party dependencies:"
-    echo "  1. GTK+, FFmpeg and GStreamer libraries used by OpenCV"
-    echo "  2. libusb library required for Myriad plugin for Inference Engine"
-    echo "  3. build dependencies for OpenVINO samples"
-    echo "  4. build dependencies for GStreamer Plugins"
-    echo
-    PKGS=(
-        cpio
-        build-essential
-        cmake
-        libusb-1.0-0-dev
-        libdrm-dev
-        libgstreamer1.0-0
+if [ "$os" == "auto" ] ; then
+    os=$( . /etc/os-release ; echo "${ID}${VERSION_ID}" )
+    case $os in
+        centos7|ubuntu18.04|ubuntu20.04) echo "Detected OS: ${os}" ;;
```

Comment:
add centos8 as well

Reply:
From my experiments, adding CentOS 8 is not trivial, because some packages have been moved to separate repositories, e.g. [flite](https://centos.pkgs.org/8/centos-powertools-x86_64/flite-1.3-31.el8.x86_64.rpm.html) and have to be handled differently. Let's implement it separately later.